Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# This script mimics a MC chain with HLT run 
+# a in a different release from the current one
+
+echo '#### MC Setup'
+
+# this would make it run only in 14_0_X releases, need to keep?
+
+# if [[ ! ("$CMSSW_VERSION" == *"14_0_"*) ]]; then
+#     echo "This is not a 14_0_X release. No need to test this. Exiting."
+#     exit 0
+# fi
+
+# inputs
+conditions=$1 
+era=$2
+hlt=$3
+release_hlt=$4
+gt_hlt=$5
+beamspot=$6
+
+base_dir=$PWD
+base_cms=$CMSSW_BASE
+base_arch=$SCRAM_ARCH
+
+
+# to take into account changes in gcc
+base_arch_no_gcc=$(echo $SCRAM_ARCH | cut -d "_" -f -2)
+if [[ ! ("$CMSSW_VERSION" == $release_hlt) ]]; then
+    hlt_arch=$(echo $(ls -dtrh /cvmfs/cms.cern.ch/${base_arch_no_gcc}*/cms/cmssw/${release_hlt} | tail -1) | cut -d "/" -f 4)
+else
+    hlt_arch=$base_arch
+fi
+
+
+echo '> GT           : ' $conditions
+echo '> Era          : ' $era
+echo '> BS           : ' $beamspot
+echo '> HLT          : ' $hlt
+echo '> HLT release  : ' $release_hlt
+echo '> HLT GT       : ' $gt_hlt
+echo ' - with SCRAM_ARCH='$hlt_arch
+echo ''
+
+
+
+############################################################################################################
+# setting up the temporary working dir, to be removed once done (or crashed)
+tmp_dir=$(mktemp -d -t test_MC_${3}_${4}_${5}_XXXXXXX)
+mkdir -p $tmp_dir
+
+echo '> Working dir: ' ${tmp_dir}
+echo ''
+
+# the tmp dir cleaning 
+function job_clean()
```

Comment:
Why is the cleanup useful? We don't generally clean up test artifacts, and in some cases having them around after a (failed) script is useful for debugging.

Reply:
All the cleanup was connected with https://github.com/cms-sw/cmssw/pull/44578#discussion_r1547337493, the idea I needed to be outside the "default" testing folder and then the fact I thought it was needed to clean up this extra repository.